File: Term 1/SE/PycharmProjects/pythonProject/DSA/linearSearch.py
#function for order linear search with input: list & key(keyword needing to found)
def linearSearch(list, key):
    #initilaise index
    i = 0
    #initialise the place that keyword at
    pos = -1

    #index inside a
    while i < len(list):
        #checking each element of the list in turn, until the desired element is found.
        if list[i] == key:
            pos = i     #position = index in a
        #continue if place haven't been found
        i += 1

    #the position found
    return pos

#initialise list from user
list = []
lenList = int(input("Enter the length of your list: "))
print("Enter the list: ")
#enter each number in list
for x in range(lenList):
    num = int(input())
    list.append(num)        #add each number to list
print(list)
key = int(input("The keyword you wanna search: "))
print(linearSearch(list, key))

File: Somophore/DSA/src/Search_algorithms/LinearSearch.java
package Search_algorithms;


public class LinearSearch {
	public int Linear_Search(int[] a, int n, int key) {
		int i = 0;
		boolean found = false;
		int pos = -1;
		while ((i < n) && (!found)) {
			if (a[i] == key) {
				found = true;
				pos = i;
				
			}i++;
		}
		return pos;
	}
	
}
